### PR TITLE
`TypedSDK` - update all `int` struct properties to `int64`

### PR DESCRIPTION
**Pull request by jackofallops**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave "+1" or "me too" comments, they generate extra noise for PR followers and do not help prioritize for review

## Description

This change updates all `int` references in the Typed resources to be `int64` to align with the values in the SDKs provided by `go-azure-sdk`

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [ ] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [ ] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [ ] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `TypedSDK` - update all `int` struct properties to `int64`

### CODE REVIEW

File: internal/services/sentinel/sentinel_threat_intelligence_indicator_resource.go [hunk 1]
```diff
@@ -435,7 +436,7 @@ func (r ThreatIntelligenceIndicator) Create() sdk.ResourceFunc {
 			props.Pattern = &patternValue
 
 			if model.Confidence != -1 {
-				props.Confidence = utils.Int32(model.Confidence)
+				props.Confidence = pointer.To(int32(model.Confidence))
```
Comment (catriona-m) [inline internal/services/sentinel/sentinel_threat_intelligence_indicator_resource.go:439]:
```suggestion
				props.Confidence = pointer.To(int64(model.Confidence))
```

File: internal/services/sentinel/sentinel_threat_intelligence_indicator_resource.go [hunk 2]
```diff
@@ -561,7 +562,7 @@ func (r ThreatIntelligenceIndicator) Update() sdk.ResourceFunc {
 				if model.Confidence == -1 {
 					properties.Confidence = nil
 				} else {
-					properties.Confidence = &model.Confidence
+					properties.Confidence = pointer.To(int32(model.Confidence))
```
Comment (catriona-m) [inline internal/services/sentinel/sentinel_threat_intelligence_indicator_resource.go:565]:
```suggestion
					properties.Confidence = pointer.To(int64(model.Confidence))
```

**Review by catriona-m — APPROVED** [review]
Apart from a couple of overlooked int32s, LGTM 🐐

File: internal/services/sentinel/sentinel_threat_intelligence_indicator_resource.go (hunk 2 above)
Comment (jackofallops) [reply internal/services/sentinel/sentinel_threat_intelligence_indicator_resource.go:565]: This is a Kermit API, so `properties.Confidence` is actually 32-bit in the API for now until it's moved over to `go-azure-sdk`.

File: internal/services/sentinel/sentinel_threat_intelligence_indicator_resource.go (hunk 1 above)
Comment (jackofallops) [reply internal/services/sentinel/sentinel_threat_intelligence_indicator_resource.go:439]: This is a Kermit API, so `properties.Confidence` is actually 32-bit in the API for now until it's moved over to `go-azure-sdk`.

**Pull request merged**